File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.java
import java.util.stream.IntStream;

class Solution {
    public int solution(int[][] board, int k) {
        return IntStream.range(0, board.length)
                        .map(i -> IntStream.range(0, board[i].length)
                                .filter(j -> i+j <= k)
                                .map(j -> board[i][j])
                                .sum()
                        )
                        .sum();
    }
}

File: 프로그래머스/0/181829. 이차원 배열 대각선 순회하기/이차원 배열 대각선 순회하기.js
function solution(board, k) {
    return Array.from({length: board.length}, (_, i) => {
        return Array.from({length: board[i].length}, (_, j) => {
                    if (i + j <= k) {
                        return board[i][j];
                    }
                    else return 0;
                })
                .reduce((a, b) => a + b, 0);
            })
            .reduce((a, b) => a + b, 0);
}